library/src/main/java/com/novoda/downloadmanager/NotificationCreator.java

```diff
@@ -1,55 +1,8 @@
 package com.novoda.downloadmanager;
 
-import android.app.Notification;
-import android.content.Context;
-import android.support.v4.app.NotificationCompat;
+interface NotificationCreator<T> {
 
-class NotificationCreator<T> {
+    void setNotificationChannelProvider(NotificationChannelProvider notificationChannelProvider);
```

Comment:
Do we really need this to be in the interface?
I don't know, it looks weird to have a setter here...

Reply:
The `interface` is internal only. If I understand the `DownloadManagerBuilder` correctly, this can't be removed. The client has the option to change the `NotificationChannel` using the `builder` pattern, after the client has already decided on the type of `customisation` they want.